yairl/discrim_net.py

```diff
@@ -90,3 +97,57 @@ def build_policy_train_reward(self):
         self._log_D, self._log_D_compl = tf.split(
             self._log_softmax_logits, [1, 1], axis=1)
         return self._log_D - self._log_D_compl
+
+
+class DiscrimNetGAIL(DiscrimNet):
+    """The discriminator to use for GAIL."""
+
+    def __init__(self, env):
+        self.env = util.maybe_load_env(env)
+
+        phs = util.build_placeholders(self.env, True)
+        self.old_obs_ph, self.act_ph, self.new_obs_ph = phs
+
+        self.log_policy_act_prob_ph = tf.placeholder(shape=(None,),
+                                                     dtype=tf.float32, name="log_ro_act_prob_ph")
+
+        with tf.variable_scope("discrim_network"):
+            self._discrim_logits = self.build_discrm_network(
+                self.old_obs_ph, self.act_ph)
+
+        super().__init__()
+
+        tf.logging.info("using GAIL")
+
+    def build_discrm_network(self, obs_input, act_input):
+        inputs = tf.concat([
+            util.flat(obs_input, self.env.observation_space.shape),
+            util.flat(act_input, self.env.action_space.shape)], axis=1)
+
+        discrim_logits = util.apply_ff(inputs, hid_sizes=[])
```

Comment:
The discriminator network is just linear in the inputs! I was surprised that this is enough, but Cartpole is a simple environment